### ISSUE TITLE
regression with markdown-in-html

### ISSUE THREAD

**Issue by parmentelat** [trentm/python-markdown2#565]
**Describe the bug**

I am currently facing a crash of markdown2 that results in this stack dump
```
Traceback (most recent call last):
  File "/<snip>/r2lab.inria.fr/md/views.py", line 565, in markdown_page
    html = markdown_module.markdown(
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/<snip>/r2lab.inria.fr/lib/python3.12/site-packages/markdown2.py", line 177, in markdown
    use_file_vars=use_file_vars, cli=cli).convert(text)
                                          ^^^^^^^^^^^^^
  File "/<snip>/r2lab.inria.fr/lib/python3.12/site-packages/markdown2.py", line 390, in convert
    text = self._do_markdown_in_html(text)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/<snip>/r2lab.inria.fr/lib/python3.12/site-packages/markdown2.py", line 987, in _do_markdown_in_html
    return self._strict_tag_block_sub(text, self._block_tags_a, callback, True)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/<snip>/r2lab.inria.fr/lib/python3.12/site-packages/markdown2.py", line 967, in _strict_tag_block_sub
    block = callback(block.rstrip('\n'))  # remove trailing newline
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/<snip>/r2lab.inria.fr/lib/python3.12/site-packages/markdown2.py", line 983, in callback
    block = self._hash_html_block_sub(block)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/<snip>/r2lab.inria.fr/lib/python3.12/site-packages/markdown2.py", line 794, in _hash_html_block_sub
    tag = tag or re.match(r'^<(\S).*?>', html).group(1)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'group'
```

**To Reproduce**
I am yet to find out the location in my - rather large - input file that causes the issue - will try to report here if I can come up with something decently small

**Expected behavior**
ideally, it would be great if
- the call would not terminate so abruptly, and would return some - possibly degraded - html
- there was some way to pinpoint where in the file (like line/col number, or any other hint) the problem actually triggered

**Debug info**
this is with 2.4.12; with 2.4.11 there is no crash

Any extras being used:

yes, a few, markdown-in-html being one, and if I take it out the problem disappears, of course

**Additional context**

for the record, the same input has been working fine for years, and has recently -not sure with which version exactly - started to misbehave, i.e. some markdown in html that used to be interpreted as markdown have ceased to be, and so show up like this e.g.
![image](https://github.com/trentm/python-markdown2/assets/4519227/680a8f19-81c7-4885-81ce-a752cedd7485)

this is clearly a separate issue, but it appears that the `markdown-in-html` thing has recently gone through changes that are breaking things for us

**Comment by parmentelat** [comment]
FWIW the original issue with the stack dump seems to have to see with HTML fragments expected to begin at the beginning of their line
I have been able to fix my inputs by sticking all the html tags to the left edge of the file
however I have not so far been able to easily build a small input that exhibits the problem...

**Comment by Crozzers** [comment]
> I have been able to fix my inputs by sticking all the html tags to the left edge of the file

And that completely solved the crash?
In that case, it would have been [this commit](https://github.com/trentm/python-markdown2/commit/f7640b137a34468c7e3c1c47569c25875e641eb5) that caused the issue, which was included in `v2.4.12`.

I managed to piece together a sample that reproduces the issue:
```
- one two
   <p markdown="1">
  def
  </p>
- three four
```
The HTML needs to be inside a list with the start tag indented slightly further than one of the other lines in the HTML.

I'll see if I can get a fix for this
